src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/config/S3Configuration.kt

```diff
@@ -0,0 +1,34 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.config
+
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials
+import software.amazon.awssdk.regions.Region
+import software.amazon.awssdk.services.s3.S3Client
+import java.net.URI
+
+@Configuration
+class S3Configuration(
+  @Value("\${aws.s3.provider}") private val provider: String,
+  @Value("\${aws.s3.region}") private val region: String,
+  @Value("\${aws.s3.access-key-id}") private val accessKeyId: String,
+  @Value("\${aws.s3.secret-access-key}") private val secretAccessKey: String,
+  @Value("\${aws.s3.endpoint.uri:}") private val uri: String,
```

Comment:
Is the ':' a typo?

Reply:
good spot! no! it's used to provide a default value if the configuration property is missing. in this case, when running locally, the URI is set, when running in production, the URI configuration is missing (and unused). the `:` defaults it to `""`. the service would not run without it.